include/ck/utility/amd_buffer_addressing.hpp

```diff
@@ -34,6 +34,21 @@ __device__ int32x4_t make_wave_buffer_resource(T* p_wave, index_t element_space_
     return wave_buffer_resource.content;
 }
 
+template <typename T>
+__device__ int32x4_t make_wave_buffer_resource_with_default_range(T* p_wave)
+{
+    BufferResource<T> wave_buffer_resource;
+
+    // wavewise base address (64 bit)
+    wave_buffer_resource.address(Number<0>{}) = const_cast<remove_cv_t<T>*>(p_wave);
+    // wavewise range (32 bit)
+    wave_buffer_resource.range(Number<2>{}) = 0xffffffff; // max possible range
```

Comment:
To reduce the number of function, I thought you can just modify make_wave_buffer_resource.
Add template parameter like this.
template <typename T, bool default_range=false>

Reply:
One thing about this approach is, I still need to pass in a `element_space_size` parameter as input. Although I can still provide this parameter with a default value, but we may result in 2 parameter both with default value, we may potentially result in unwanted behavior. So here I suggest keep it this way.